### PR TITLE
Feat/317/verbose /admin announce summary

### LINKED ISSUE #317: verbose output from praise bot when sending mass messages
currently when I send a mass DM to quantifiers this is the output:

image

I can see that it wasn't delivered to one user but not which user?? This is kind of important because without knowing who wasn't notified the only sure way to notify all quantifiers is for me to DM everybody like I was doing before.

It would be great to have more information given to me from sending these messages, such as who was notified and who wasn't

### PR DESCRIPTION
**Pull request by Vyvy-vi**
Resolves #317 

This PR adds a summary report file that is sent when the /admin announce command is used. We might need to refactor this a bit in future and use the praise username instead of discord ids.

Screenshot 2022-06-04 at 12 27 33 AM
Screenshot 2022-06-04 at 12 27 49 AM

### CODE REVIEW

File: packages/discord-bot/src/utils/dmTargets.ts [hunk 1]
```diff
@@ -49,9 +50,31 @@ const sendDMs = async (
       : failed.length === 0
       ? successMsg
       : successMsg + '\n' + failedMsg;
+
+  // TODO - Create a utility function to tabularise this data neatly
+  let summary = `User\t\t\t\tStatus\n`;
```
Comment (mattyg) [inline packages/discord-bot/src/utils/dmTargets.ts:55]: lint warning

**Review by mattyg — CHANGES_REQUESTED** [review]
Awesome!

I think we need to replace the discord ids with user names before it effectively resolves #317.

**Comment by Vyvy-vi** [comment]
> I think we need to replace the discord ids with user names before it effectively resolves #317.
This can be resolved when #377 is implemented. We can't use discord usernames if the user isn't present in the discord.
One temp. solution may be to fetch all the `UserAccount`s associated with the `User` and to return a more detailed output -

| User | DiscordUsername | TelegramUsername | Status | Reason |
| :-: | :-: | :-: | :-: | :-: |
| A | A_on_discord_old_account | A_on_Telegram | Undelivered | User Not Found in discord |
| B | B_on_discord | B_on_Telegram | Delivered | - | 
| C | C_on_discord | C_on_Telegram | Undelivered | Can't DM User | 

Would this add a significant overhead to the process? 

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by kristoferlund** [comment]
> This can be resolved when #377 is implemented. We can't use discord usernames if the user isn't present in the discord.

How could the user not be on Discord? In order to be a quantifier you need to activate your account, linking `UserAccount` to `User`.

**Comment by Vyvy-vi** [comment]
> How could the user not be on Discord? In order to be a quantifier, you need to activate your account, linking `UserAccount` to `User`.

People can leave the discord server after activating, or they can be kicked/banned for some reason, or their discord account might have been deleted.

**Comment by Vyvy-vi** [comment]
That's true. It might add some extra API calls, but we can add that 👍 

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by Vyvy-vi** [comment]
The code isn't as DRY as I'd like, however, I think we can work on that when we make a separate function for tabularisation (or if we opt to use a CSV file, instead).

Here's the new version -
Screenshot 2022-06-11 at 12 59 17 AM
Screenshot 2022-06-11 at 12 58 57 AM

**Comment by Vyvy-vi** [comment]
also, how to type this properly? -
```
interface FailedToDmUsersList {
  invalidUsers: string[];
  notFoundUsers: string[];
  closedDmUsers: string[];
  unknownErrorUsers: string[];
}

const failed: FailedToDmUsersList = {
  invalidUsers: <string[]>[],
  notFoundUsers: <string[]>[],
  closedDmUsers: <string[]>[],
  unknownErrorUsers: <string[]>[]
};
```

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by mattyg** [comment]
> also, how to type this properly? -

Since you already defined it in the interface, you can just create an object that implements it. The type `string[]` allows for an empty array (as long as any items you subsequently add to it are strings). So like this:

```
interface FailedToDmUsersList {
  invalidUsers: string[];
  notFoundUsers: string[];
  closedDmUsers: string[];
  unknownErrorUsers: string[];
}

const failed: FailedToDmUsersList = {
  invalidUsers: [],
  notFoundUsers: [],
  closedDmUsers: [],
  unknownErrorUsers: []
};
```

**Review by mattyg — APPROVED** [review]
awesome!

**Review by kristoferlund — APPROVED** [review]
Looks great @Vyvy-vi! I made two small text changes. Now merging.

**Pull request merged**